Mathlib/Algebra/Algebra/Subalgebra/Basic.lean

```diff
@@ -366,72 +361,60 @@ instance : Module R S :=
   S.module'
 
 instance [Semiring R'] [SMul R' R] [Module R' A] [IsScalarTower R' R A] : IsScalarTower R' R S :=
-  S.toSubmodule.IsScalarTower
+  inferInstanceAs (IsScalarTower R' R (toSubmodule S))
 
 instance algebra' [CommSemiring R'] [SMul R' R] [Algebra R' A] [IsScalarTower R' R A] :
     Algebra R' S :=
-  {
-    (algebraMap R' A).codRestrict S fun x =>
-      by
+  { (algebraMap R' A).codRestrict S fun x => by
       rw [Algebra.algebraMap_eq_smul_one, ← smul_one_smul R x (1 : A), ←
         Algebra.algebraMap_eq_smul_one]
-      exact
-        algebra_map_mem S
+      exact algebraMap_mem S
           _ with
     commutes' := fun c x => Subtype.eq <| Algebra.commutes _ _
     smul_def' := fun c x => Subtype.eq <| Algebra.smul_def _ _ }
 #align subalgebra.algebra' Subalgebra.algebra'
 
-instance : Algebra R S :=
-  S.algebra'
+instance : Algebra R S := S.algebra'
 
 end
 
 instance noZeroSMulDivisors_bot [NoZeroSMulDivisors R A] : NoZeroSMulDivisors R S :=
-  ⟨fun c x h =>
-    have : c = 0 ∨ (x : A) = 0 := eq_zero_or_eq_zero_of_smul_eq_zero (congr_arg coe h)
+  ⟨fun {c} {x : S} h =>
+    have : c = 0 ∨ (x : A) = 0 := eq_zero_or_eq_zero_of_smul_eq_zero (congr_arg Subtype.val h)
```

Comment:
Strictly `(↑)` would be a more faithful port of `coe`, but it doesn't matter

Reply:
It also doesn't work.